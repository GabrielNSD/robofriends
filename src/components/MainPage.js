import React, { useState, useEffect } from "react";
import CardList from "./CardList";
import SearchBox from "./SearchBox";
import Scroll from "./Scroll";
import "./MainPage.css";
import ErrorBoundry from "./ErrorBoundry";
import CounterButton from "./CounterButton";

import Header from "./Header";

const MainPage = (props) => {
  const { searchField, onSearchChange, robots, isPending } = props;

  useEffect(() => {
    props.onRequestRobots();
  }, []);

  const filteredRobots = robots.filter((robot) => {
    return robot.name.toLowerCase().includes(searchField.toLowerCase());
  });

  return isPending ? (
    <h1>Loading</h1>
  ) : (
    <div className="tc">
      <Header />
      <CounterButton />
      <SearchBox searchChange={onSearchChange} />
      <Scroll>
        <ErrorBoundry>
          <CardList robots={filteredRobots} />
        </ErrorBoundry>
      </Scroll>
    </div>
  );
};

export default MainPage;
