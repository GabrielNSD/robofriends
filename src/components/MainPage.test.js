import React from "react";
import { shallow, mount, render } from "enzyme";
import { create } from "react-test-renderer";
import MainPage from "./MainPage";

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: "",
    isPending: false,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});

it("renders MainPAge without crashing", () => {
  expect(wrapper).toMatchSnapshot();
});

it('filters robots correctly', () => {
  const test = wrapper.find("CardList")
  console.log(test)
  expect(wrapper).toEqual({})
})