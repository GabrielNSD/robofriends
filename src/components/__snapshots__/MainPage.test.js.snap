// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPAge without crashing 1`] = `
<div
  className="tc"
>
  <Memo() />
  <CounterButton />
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <CardList
        robots={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
